server/routes/referrals/referralFormPresenter.ts

```diff
@@ -1,118 +1,123 @@
-    return this.serviceCategories.length > 1
+    return this.intervention.serviceCategories.length > 1
   }
 
   get sections(): ReferralFormSectionPresenter[] {
-    const reviewServiceUserInformation: ReferralFormSingleListSectionPresenter = {
-      type: 'single',
-      title: 'Review service user’s information',
-      number: '1',
-      status: this.calculateStatus(this.sectionValues.reviewServiceUserInformation),
-      tasks: [
-        {
-          title: 'Confirm service user’s personal details',
-          url: 'service-user-details',
-        },
-        {
-          title: 'Service user’s risk information',
-          url: this.calculateTaskUrl('risk-information', this.taskValues.serviceUserDetails),
-        },
-        {
-          title: 'Service user’s needs and requirements',
-          url: this.calculateTaskUrl('needs-and-requirements', this.taskValues.riskInformation),
-        },
-      ],
-    }
-    let referralDetails: ReferralFormSectionPresenter
     if (this.isCohortIntervention) {
-      referralDetails = {
-        type: 'multi',
-        title: 'Add intervention referral details',
+      return this.formSectionBuilder.buildCohortReferralSections()
+    }
+    return this.formSectionBuilder.buildSingleReferralSections()
+  }
+
+  private static FormSectionBuilder = class {
+    constructor(
+      private referral: DraftReferral,
+      private intervention: Intervention,
+      private taskValues: typeof ReferralFormPresenter.TaskValues.prototype,
+      private sectionValues: typeof ReferralFormPresenter.SectionValues.prototype
+    ) {}
+
+    buildSingleReferralSections(): ReferralFormSectionPresenter[] {
+      const reviewServiceUserInformationSection = this.buildReviewServiceUserInformationSection()
+      const referralDetailsSection = this.buildSingleReferralDetailsSection(reviewServiceUserInformationSection)
+      const checkYourAnswersSection = this.buildCheckYourAnswersSection(referralDetailsSection, false)
+      return [reviewServiceUserInformationSection, referralDetailsSection, checkYourAnswersSection]
+    }
+
+    buildCohortReferralSections() {
+      const reviewServiceUserInformationSection = this.buildReviewServiceUserInformationSection()
+      const selectServiceCategoriesSection = this.buildSelectServiceCategoriesSection(
+        reviewServiceUserInformationSection
+      )
+      const referralDetailsSection = this.buildCohortReferralDetailsSection(selectServiceCategoriesSection)
+      const checkYourAnswersSection = this.buildCheckYourAnswersSection(referralDetailsSection, true)
+      return [
+        reviewServiceUserInformationSection,
+        selectServiceCategoriesSection,
+        referralDetailsSection,
+        checkYourAnswersSection,
+      ]
+    }
+
+    private buildReviewServiceUserInformationSection(): ReferralFormSingleListSectionPresenter {
+      return {
+        type: 'single',
+        title: 'Review service user’s information',
+        number: '1',
```

Comment:
what is 'number' used for ?

also a small code smell: a field called 'number' that's a string?

Reply:
This number is what is displayed for the section header, i.e.:
```
1. Review service user's information
```

I agree that it would be better if it was done with css. This was the implementation before I touched on changing it for cohort referrals; would you like me to make this change as well?

Reply:
ah i see - no it's ok. the main reason i noticed this was the conditional 3?4 for cohort referrals. seems a little flakey. let's leave for now and change to auto numbering the sections later.